dags/iop/parser.py

```diff
@@ -267,3 +282,55 @@ def _get_copyright(self, article):
         if extracted_statement:
             copyright["copyright_statement"] = extracted_statement
         return [copyright]
+
+    def _get_license_type_and_version(self, license_node, url):
+        try:
+            license_type = license_node.get("license-type")
+            version = LICENSE_VERSION_PATTERN.search(url).group(0)
+            return construct_license(
+                license_type=license_type.upper(), version=version, url=url
+            )
+        except AttributeError:
+            return
+
+    def _get_license_from_text(self, license_node):
+        license_text = extract_text(
+            path="license-p/ext-link",
+            article=license_node,
+            field_name="ext-link",
+            dois=self.dois,
+        )
+        license_type = get_license_type(license_text=license_text)
+        version = LICENSE_VERSION_PATTERN.search(license_text).group(0)
+        return construct_license(license_type=license_type, version=version)
+
+    def _get_license(self, article: ET.Element):
+        licenses = []
+        licenses_nodes = article.findall("front/article-meta/permissions/license")
+        for license_node in licenses_nodes:
+            try:
+                url = license_node.attrib["{http://www.w3.org/1999/xlink}href"]
+                type_and_version = self._get_license_type_and_version(
+                    license_node=license_node, url=url
+                )
+                if type_and_version:
+                    licenses.append(type_and_version)
+                else:
+                    self.logger.error(
+                        "License type is not found in XML. Parsing from URL..."
+                    )
+                    type_and_version = get_license_type_and_version_from_url(url=url)
+                    if type_and_version:
+                        licenses.append(type_and_version)
+            except (KeyError):
+                self.logger.error(
+                    "License URL is not found in XML or unknown license found. Parsing from license text..."
+                )
+                type_and_version = self._get_license_from_text(
+                    license_node=license_node
+                )
+                if type_and_version:
+                    licenses.append(type_and_version)
+                else:
+                    self.logger.error("License is not found in license text.")
+        return licenses
```

Comment:
@ErnestaP could you please write each case of licenses? I think this is too complicated for what we actually need.

Reply:
Could you please also give me some examples from the actual records?